File: src/go/shipments/buy-shipment.go
package main

import (
	"fmt"
	"os"
	"encoding/json"
	"github.com/EasyPost/easypost-go"
)

func main() {
	apiKey := os.Getenv("EASYPOST_API_KEY")
	if apiKey == "" {
		fmt.Fprintln(os.Stderr, "missing API key")
		os.Exit(1)
		return
	}
	client := easypost.New(apiKey)

	// buy postage label with one of the rate objects
	// client.BuyShipment(shipment, rate, insurance)
	shipment, err := client.BuyShipment("shp_a52895d6f5a141ec87bac56e73825330", &easypost.Rate{ID: "rate_588b263a7a9449c986efdcf01b7821c7"}, "")
	if err != nil {
		fmt.Fprintln(os.Stderr, "error buying shipment:", err)
		os.Exit(1)
		return
	}

	prettyJSON, err := json.MarshalIndent(shipment, "", "    ")
    if err != nil {
		fmt.Fprintln(os.Stderr, "error creating JSON:", err)
    }
	fmt.Printf("%s\n", string(prettyJSON))
}

File: src/python/shipments/buy_shipment.py
import os

import easypost
from dotenv import load_dotenv

load_dotenv()
easypost.api_key = os.getenv('EASYPOST_TEST_API_KEY')


try:
    shipment = easypost.Shipment.retrieve('shp_3cba0177d3ca4bb9a00fedb9e5bc2cae')
    response = shipment.buy(rate={'id': 'rate_4ee872c4755d4af09ec91be0f4404c85'})
    print(response)
except Exception as error:
    print(error)
